src/Smearing.jl

```diff
@@ -75,14 +71,32 @@ function occupation_derivative(S::FermiDirac, x)
         -exp(x) / (1+exp(x))^2
     end
 end
-
 # entropy(f) = -(f log f + (1-f)log(1-f)), where f = 1/(1+exp(x))
 # this "simplifies" to -(x*exp(x)/(1+exp(x)) - log(1+exp(x)))
 # although that is not especially useful...
+function xlogx(x)
+    iszero(x) ? zero(x) : x * log(x)
+end
 function entropy(S::FermiDirac, x)
     f = occupation(S, x)
     - (xlogx(f) + xlogx(1 - f))
 end
+function occupation_divided_difference(S::FermiDirac, x, y, εF, temperature)
+    temperature == 0 && return occupation_divided_difference(None(), x, y, εF, temperature)
+    f(z) = occupation(S, (z-εF) / temperature)
+    fder(z) = occupation_derivative(S, (z-εF)/temperature) / temperature
+    # For a stable computation we use
+    # (fx - fy) = fx fy (exp(y) - exp(x)) = fx fy exp(x) expm1(y-x)
+    # which we symmetrize. This can overflow, in which case we fall back to the standard method
+    will_exp_overflow(z1, z2) = abs((z1-z2)/temperature) > log(floatmax(typeof(x))) / 4 # conservative
```

Comment:
You know how I feel about long lines ...

Reply:
There's no logical place to break here

Reply:
come on, just assign `floatmax(typeof(x))` to something, or `T = typeof(x)`

Reply:
less of a problem now that temperature is smaller